### PR TITLE
Add OpenCL Demo

### PR DESCRIPTION
**Pull request by LonerDo**

### CODE REVIEW

**Comment by LonerDo** [comment]
@dotnet-policy-service agree

File: examples/CSharp/OpenCL Tutorials/HelloWorld/HelloWorld.csproj [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (Perksey) [inline examples/CSharp/OpenCL Tutorials/HelloWorld/HelloWorld.csproj:1]: Can you move this project to the `OpenCL Demos` folder until we figure out what a formal set of tutorials looks liek for OpenCL?

File: examples/CSharp/OpenCL Tutorials/HelloWorld/Program.cs [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+        /// <param name="cl"></param>
+        /// <param name="context"></param>
+        /// <param name="device"></param>
+        /// <param name="fileName"></param>
+        /// <returns></returns>
+        static unsafe nint CreateProgram(CL cl, nint context, nint device, string fileName)
+        {
+            if (!File.Exists(fileName))
+            {
+                Console.WriteLine($"文件不存在:{fileName}");
```
Comment (Perksey) [inline examples/CSharp/OpenCL Tutorials/HelloWorld/Program.cs:166]:
```suggestion
                Console.WriteLine($"File does not exist: {fileName}");
```

File: examples/CSharp/OpenCL Tutorials/HelloWorld/Program.cs [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+            return commandQueue;
+        }
+
+        /// <summary>
+        /// Create an OpenCL context on the first available platform using
+        /// either a GPU or CPU depending on what is available.
+        /// </summary>
+        /// <param name="cL"></param>
+        /// <returns></returns>
+        static unsafe nint CreatContext(CL cL)
```
Comment (Perksey) [inline examples/CSharp/OpenCL Tutorials/HelloWorld/Program.cs:289]:
```suggestion
        static unsafe nint CreateContext(CL cL)
```
(and rename usages)

**Review by Perksey — CHANGES_REQUESTED** [review]
This looks really good, thank you so much for your amazing contribution! Just a few minor nits, most of this I couldn't do better if I tried!

**Review by Perksey — APPROVED** [review]
Awesome, thank you so much for your contribution!

**Pull request merged**